src/mono/CMakeLists.txt

```diff
@@ -655,6 +655,21 @@ endif()
 ######################################
 # EXTRACT VERSION
 ######################################
+if (NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/_version.h")
+  file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/_version.h" "static char sccsid[] __attribute__((used)) = \"@(#)Version 42.42.42.42424 @Commit: AAA\";\n")
+endif()
+if (NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/runtime_version.h")
+  file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/runtime_version.h"
+"#define RuntimeFileMajorVersion 42\n"
+"#define RuntimeFileMinorVersion 42\n"
+"#define RuntimeFileBuildVersion 42\n"
+"#define RuntimeFileRevisionVersion 42424\n"
+"#define RuntimeProductMajorVersion 6\n"
+"#define RuntimeProductMinorVersion 0\n"
+"#define RuntimeProductPatchVersion 0\n"
+)
+endif()
```

Comment:
instead of writing to these unused header files and reading them back I'd rather do

```cmake
if (NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/_version.h" AND NOT EXISTS "${CMAKE_CURRENT_BINARY_DIR}/runtime_version.h")
  set(VERSION "\"42.42.42.0\"")
  set(FULL_VERSION "42.42.42.42424")
else()
 ... existing code

endif()
```

Reply:
runtime-version.h is included in some eventpipe files.